packages/iov-bcp-types/src/bcp.ts

```diff
@@ -70,15 +88,25 @@ export interface BcpValueNameQuery {
 
 export type BcpAccountQuery = BcpAddressQuery | BcpValueNameQuery;
 
-// IovReader is a high-level interface to a blockchain node,
+// isQueryByAddress is a type guard to use in the account-based queries
+export function isQueryByAddress(query: BcpAccountQuery): query is BcpAddressQuery {
+  return (query as BcpAddressQuery).address !== undefined;
+}
+
+// BcpConnection is a high-level interface to a blockchain node,
 // abstracted over all blockchain types and communication channel.
 // A direct connection or a proxy server should implement this.
 // The implementation takes care to convert our internal types into
 // the proper format for the blockchain.
 //
-// The IovReader will most likely contain some private state to maintain
-// the connection, subscription, and such.
-export interface IovReader {
+// BcpConnection is the minimal interface needed to be supported by any blockchain
+// that is compatible with the bcp spec and iov-core library. This supports
+// getting account balances, sending tokens, and observing the blockchain state.
+//
+// There are other optional interfaces that extend this functionality with
+// features like atomic swap, NFTs, etc which may be implemented by any connector
+// to enable enhanced features in the clients
+export interface BcpConnection {
```

Comment:
This should not break anything in client side if its still referred to as a `writer.reader(chainId)`

Reply:
So far I just changed interfaces, not method names and not class names.

I would like to properly update writer, but that is another pr as it is far more breaking that this feature

Reply:
Ok, I will wait for the writer update so I can be prepared to update my end to match when the time comes.

Reply:
I would do that after swap and lisk are merged, so we can tag something useful for iov-wallet before breaking stuff.

I would also like to update redux to typescript, as it is much easier to handle renamings with typescript, as you get compile errors or warnings in the ide if the methods don't match. Otherwise it is a bunch of hunting around